src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,33 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        return CompletableFuture
+                .supplyAsync(
+                        () -> cache.computeIfAbsent(userId, this::getUserData),
+                        executor
+                );
+    }
+
+    private UserData getUserData(String userId) {
+        try {
+            TimeUnit.SECONDS.sleep(1);
```

Comment:
The sleep here simulates a time-consuming operation, which is fine for an example. However, in a production environment, this should be replaced with actual data fetching logic, such as a database call or a web service request. The sleep operation blocks the current thread, which could be a waste of resources if this is meant to simulate I/O-bound work.